### PR TITLE
disable review button for validation errors on fields

### PR DESCRIPTION
**Pull request by Nil20**
Farajaland PR: https://github.com/opencrvs/opencrvs-farajaland/pull/791

### CODE REVIEW

**Comment by Nil20** [comment]
[FieldAgentDisable.webm](https://github.com/opencrvs/opencrvs-core/assets/32668488/ed1e4220-7c56-4c92-9df7-c6d670ec8a0b)

File: packages/client/src/components/form/ReviewActionComponent.tsx [hunk 1]
```diff
@@ -377,7 +379,7 @@ class ReviewActionComponent extends React.Component<
                 size="large"
                 id="submit_form"
                 onClick={this.toggleSubmitModalOpen}
-                disabled={totalFileSizeExceeded}
+                disabled={!hasErrorsOnFields || totalFileSizeExceeded}
```
Comment (rikukissa) [inline packages/client/src/components/form/ReviewActionComponent.tsx:382]: This variable is a little confusing. The submit button is disabled if there are no errors on the fields? Shouldn't it be the other way around?

File: packages/client/src/views/RegisterForm/review/ReviewSection.tsx [hunk 2]
```diff
@@ -1977,6 +1990,7 @@ class ReviewSectionComp extends React.Component<FullProps, State> {
                         declaration={declaration}
                         submitDeclarationAction={submitClickEvent}
                         rejectDeclarationAction={rejectDeclarationClickEvent}
+                        hasErrorsOnFields={hasValidationErrors}
```
Comment (rikukissa) [inline packages/client/src/views/RegisterForm/review/ReviewSection.tsx:1993]: I'm testing this and have an invalid date for child's DoB but this value is `false`. If I fix the error it's `true` 🤔

**Review by rikukissa — CHANGES_REQUESTED** [review]

File: packages/client/src/views/RegisterForm/review/ReviewSection.tsx (hunk 2 above)
Comment (Nil20) [reply packages/client/src/views/RegisterForm/review/ReviewSection.tsx:1993]: Yeah, fixed it on the latter commit

File: packages/client/src/i18n/messages/views/review.ts [hunk 3]
```diff
@@ -191,7 +191,7 @@ const messagesToDefine = {
   },
   reviewActionDescriptionIncomplete: {
     defaultMessage:
-      'By sending this incomplete declaration, there will be a digital record made.\n\nTell the informant that they will receive an {deliveryMethod} with a tracking ID. They will need this to complete the declaration at a registration office within 30 days. The informant will need to provide all mandatory information before the {eventType, select, birth {birth declaration} death {death declaration}} can be registered.',
+      'Please ensure all fields are either empty or have a valid value in them.',
```
Comment (Zangetsu101) [inline packages/client/src/i18n/messages/views/review.ts:194]: Is this intentional? The copy change seems pretty drastic

Comment (Nil20) [reply packages/client/src/i18n/messages/views/review.ts:194]: Yeah, this is, as discussed with @rikukissa

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**